Tasks/MavenV3/maventask.ts

```diff
@@ -538,4 +539,9 @@ function processMavenOutput(data) {
     }
 }
 
-execBuild();
+if (restoreOriginalPomXml) {
+    const originalPomContents: string = fs.readFileSync(mavenPOMFile, 'utf8');
+    execBuild().then(() => fs.writeFileSync(mavenPOMFile, originalPomContents));
```

Comment:
Please check error handling here also - how verbose message is when restoring fails.